.github/workflows/ci.yml

```diff
@@ -269,11 +273,13 @@ jobs:
           CARGO: cross
           CARGO_BUILD_TARGET: aarch64-unknown-linux-gnu
           PYO3_CROSS_LIB_DIR: /opt/python/cp38-cp38/lib
+          DIST_EXTRA_CONFIG: /tmp/build-opts.cfg
         run: |
           cd examples/namespace_package
           docker build -t cross-pyo3:aarch64-unknown-linux-gnu .
-          python -m pip install wheel
-          python setup.py bdist_wheel --plat-name manylinux2014_aarch64
+          python -m pip install build wheel
+          echo -e "[bdist_wheel]\nplat_name=manylinux2014_aarch64" > $DIST_EXTRA_CONFIG
+          python -m build --no-isolation
```

Comment:
Maybe the changes for `test-cross` and `test-crossenv` can be reverted and left for now to ease your pain in CI? Given `rust_with_cffi/setup.py` still exists I'd be ok with leaving those as-is in this PR and dealing with them another time.

Reply:
Hi @davidhewitt, thank you very much for the review.

I can try to do that. For `test-crossenv` it should be just a matter of reverting 1d3fe5f32f95ac766bf4a53b67e85ef5e87674f8. Although, I am not sure how the problem with the failures in the CI are related to `setup.py` vs. `DIST_EXTRA_CONFIG` as a form of passing build options:

- In https://github.com/PyO3/setuptools-rust/actions/runs/5834731386/job/15824909577#step:3:232 we can see that the error in CI for test-crossenv seems unrelated (somehow it is missing `decimal`, which is supposedly happening before `build` creates an isolated environment.)

- In https://github.com/PyO3/setuptools-rust/actions/runs/5834731386/job/15824908490#step:7:488), we can see that Rust seems to be having problems to find Python information and failing with:
  ```python
  error: no Python 3.x interpreter found
  ```

Maybe that is related to the use of a virtual environment in https://github.com/PyO3/setuptools-rust/actions/runs/5834731386/job/15824908490#step:7:6?

I might try a couple of things before, but if they don't work I will proceed with the suggestion of sticking with `python setup.py` for the problematic tests.

Reply:
There is something weird going on with `test-crossenv` when I try to use `build`:

```bash
+ pip install cffi
Requirement already satisfied: cffi in ./venv/build/lib/python3.9/site-packages (1.15.1)
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Installing collected packages: pycparser
Successfully installed pycparser-2.21

Notice:  A new release of pip is available: 23.0.1 -> 23.2.1
Notice:  To update, run: pip install --upgrade pip
+ export DIST_EXTRA_CONFIG=/tmp/build-opts.cfg
+ DIST_EXTRA_CONFIG=/tmp/build-opts.cfg
+ echo -e '[bdist_wheel]\npy_limited_api=cp37'
+ python -m build --no-isolation
* Getting build dependencies for sdist...

ERROR Missing dependencies:
	cffi
```

I do `pip install cffi` and it says `Requirement already satisfied`, then I try to run `python -m build --no-isolation` and it complains about `cffi` being a missing dependency.

Reply:
`crossenv` seems to have a problem with `build`: https://github.com/benfogle/crossenv/issues/108